lib/sentry/client_error.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Sentry.ClientError do
+  @moduledoc """
+  An exception struct that represents an error returned by Sentry.
+
+  This struct is designed to manage and handle errors originating from operations
+  in the client application. The `:reason` field signifies the cause of the error
+  as an atom or tuple.
+
+  To raise instances of this exception, you can use `Kernel.raise/1`. When crafting
+  formatted error messages for purposes such as logging or presentation, consider
+  leveraging `Exception.message/1`.
+  """
+
+  @moduledoc since: "10.7.0"
+
+  @doc """
+  The exception struct for a Sentry error.
+  """
+  defexception [:reason]
+
+  @typedoc """
+  The type for a Sentry error exception.
+  """
+  @type t() :: %__MODULE__{reason: term}
```

Comment:
Nit, but can we introduce

```elixir
@type reason() :: :too_many_retries | ...
```

and use that in `new/1` as well, instead of `term/0`?

Reply:
like this:

```elixir
  @type t() :: %__MODULE__{
          reason:
            :too_many_retries
            | {:invalid_json, Exception.t() / 0}
            | {:request_failure, String.t()}
            | {:request_failure, atom}
            | {:request_failure, {atom(), term(), [term()]}}
        }```

Reply:
yep!